File: 프로그래머스/unrated/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.java
class Solution {
    public int solution(int n) {
        int answer = 0;

        boolean isOdd = n % 2 == 0;

        if (isOdd) {

            for (int i = 2; i <= n; i+=2) {

                answer += i*i;
            }

        } else {

            for (int i = 1; i <= n; i+=2) {

                answer += i;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/0/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.kt
import kotlin.math.pow

class Solution {
	fun solution(n: Int): Int {
		var answer: Int = 0

		return if (n % 2 == 0) {
			(n downTo 1 step 2).sumOf() { it.toDouble().pow(2).toInt() }
		} else {
			(n downTo 1 step 2).sum()
		}
	}
}